src/preprocessing/preprocessing.rs

```diff
@@ -139,6 +139,9 @@ impl<D: Domain> PreprocessingExecution<D> {
                 Instruction::Branch(dst) => {
                     self.masks.set(dst, self.shares.branch.next());
                 }
+                Instruction::Const(_dst, _c) => {
+                    // We don't need to mask constant inputs because the circuit is public
```

Comment:
You will need a mask though, once you want to use the constant in the next gate it will need some mask. But I agree it can be zero:
`self.masks.set(dst, D::Sharing::ZERO);`
(same goes for next two files as well of course)

Reply:
That's a very good point, thanks!